src/meta/meta_ckpt.c

```diff
@@ -302,6 +306,22 @@ __wt_meta_ckptlist_get(WT_SESSION_IMPL *session, const char *fname, WT_CKPT **ck
     /* Sort in creation-order. */
     __wt_qsort(ckptbase, slot, sizeof(WT_CKPT), __ckpt_compare_order);
 
+    if (update) {
```

Comment:
I'm not so sure about this.
I haven't added the rest of the code in this `update` check because it was added to support standalone importing of WiredTiger files (which v4.0 doesn't support). But we definitely need to update the timestamp here.

Reply:
Yes, this makes sense.

But...

I was worried that you had to initialize the `ckpt->sec` field here.  Since it exists in the 4.0 code, I figured it must be getting set someplace else and went looking to see if that would cause problems.  It turns out that 4.0 sets it (along with the other initialization of a few other fields in the new `ckpt` structure) in `wt_meta_ckptlist_set()`.  So as it stands, that value would get overwritten and no longer match what we store here in `conn->ckpt_most_recent`.  

So I think we also need to remove the code that sets `ckpt->sec` in `wt_meta_ckptlist_set()`.

It would be cleaner, of course, to do all the `ckpt` initialization in one place.  But as you've probably figured out, we need to have `ckpt->sec` set before we decide what checkpoints to delete in `ckpt_lock_dirty_tree()`.  So it can't wait until `wt_meta_ckptlist_set()`, which gets called later.  And I figure that for a backport we shouldn't move the rest of `ckpt` initialization to `wt_meta_ckptlist_get()` as it increases the risk of unintended side-effects.  

Obviously a comment will help here. 😃

Reply:
e6b28fb

Thanks for picking up on that. I had a closer look at what's happening with `ckpt_lock_dirty_tree` and it's making more sense.

As you suggested, I've removed the assignment in the `set` helper and added a comment in the `get` helper explaining why the initialization is done there.

Reply:
Looks good.  Thanks!